### DISCUSSION TITLE
How to assign IP addresses to switches

### DISCUSSION THREAD

**Discussion by dreng** [netbox-community/netbox#6509] [Q&A] [answered]
Hello there,

I'd like to assign an IP address to a switch (for switch management purposes). In netbox, normally one assigns an IP to an interface. But what is the recommended way to assign an IP to a switch that has 48 interfaces? The IPs of our switches are not bound to a specific Interface but are accessable on all interfaces.

Thank you!

**Comment by jeremystretch** [comment]
There's a virtual interface on the switch to which the routed address is bound. It varies by platform, but it's definitely there. That's where the IP(s) are assigned.

What platform is this?

**Reply by dreng** [reply]
Don't know what the platform is called. Mainly Cisco SG series and a few cheap HPEs.

So, you suggest to add an interface (no. 49) to an 48-port switch and give it the type virtual. Is this correct?

**Reply by candlerb** [reply]
Yes.  There may be a direct interface name from the config like "vlan1"; if not then a generic name like "mgmt" is fine.  (Note: this is not the same as a "management-only" interface, which is used for an out-of-band management port with direct access to the control plane)

**Comment by lowinger42** [comment, marked as answer]
On 2021-05-31 21:28, Mario F. wrote:
>
> Don't know what the platform is called. Mainly Cisco SG series and a few cheap HPEs.
>
> So, you suggest to add an interface (no. 49) to an 48-port switch and give it the type virtual. Is this correct?
>
I usually create a virtual interface called "management" and assign the address to this interface.

/AL

**Comment by dreng** [comment]
That helped a lot. Thank you all.
